### PR TITLE
[KEYCLOAK-9573] Allow AdminEvents for custom resource types

### PR DESCRIPTION
**Pull request by sebastianloesch**

### CODE REVIEW

**Review by stianst — DISMISSED** [review]

**Comment by stianst** [comment]
@hmlnarik @mposolda it's not super elegant, but I can't think of a better way to do this. WDYT?

**Comment by stianst** [comment]
There's a test failure here: AdminEventStoreProviderTest.handleCustomResourceTypeEvents Uncaught server error detected: java.lang.IllegalArgumentException: No enum constant org.keycloak.events.admin.ResourceType.my-custom-resource

**Comment by sebastianloesch** [comment]
Updated the PR and fixed the test.

File: server-spi-private/src/main/java/org/keycloak/events/admin/AdminEvent.java [hunk 1]
```diff
@@ -144,10 +146,26 @@ public void setError(String error) {
      * @return
      */
     public ResourceType getResourceType() {
-        return resourceType;
+        return Arrays.stream(ResourceType.values()).filter(r -> r.toString().equals(resourceType)).findFirst().orElse(null);
```
[the remark this answers is not in the document]
Comment (sebastianloesch) [reply server-spi-private/src/main/java/org/keycloak/events/admin/AdminEvent.java]: Sounds good to me, but introducing a  `CUSTOM` constant for `ResourceType` breaks backward compatibility. Is that okay?

[the remark this answers is not in the document]
Comment (hmlnarik) [reply server-spi-private/src/main/java/org/keycloak/events/admin/AdminEvent.java]: Absolutely - the modifications are in private server SPI package which should not be used publicly.

[the remark this answers is not in the document]
Comment (sebastianloesch) [reply server-spi-private/src/main/java/org/keycloak/events/admin/AdminEvent.java]: @hmlnarik : I updated the PR with your suggestion.

[the remark this answers is not in the document]
Comment (hmlnarik) [reply server-spi-private/src/main/java/org/keycloak/events/admin/AdminEvent.java]: @sebastianloesch Thank you

**Comment by hmlnarik** [comment]
@keycloak-ci-bot test

**Review by hmlnarik — APPROVED** [review]
LGTM, thank you!

**Comment by sebastianloesch** [comment]
@hmlnarik Is there anything to do for me here on this PR?

**Review by mposolda — APPROVED** [review]
Agree that it's not super-elegant, but can't think of any nicer backwards-compatible way..

**Comment by sebastianloesch** [comment]
@mposolda: is there anything left to do to get this PR merged? Who is merging?

**Comment by mposolda** [comment]
@sebastianloesch PR is assigned to @stianst, so he will need to approve and merge.

**Comment by sebastianloesch** [comment]
@stianst: When do you expect this PR to be merged? I am really looking forward to get this feature. Best regards, Sebastian

**Comment by sebastianloesch** [comment]
@stianst Ping!

**Review by stianst — APPROVED** [review]

**Pull request merged**